src/rate_limiter.cc

```diff
@@ -319,6 +317,19 @@ RateLimiter::GetPayload(
 void
 RateLimiter::PayloadRelease(std::shared_ptr<Payload>& payload)
 {
+  // If this is an exit payload, the instance must not be staged once marked as
+  // available, so mark the instance as removing.
+  if (payload->GetOpType() == Payload::Operation::EXIT) {
+    std::lock_guard<std::mutex> lk(model_instance_ctx_mtx_);
+    auto& instances = model_instance_ctxs_[payload->GetInstance()->Model()];
+    auto it = instances.find(payload->GetInstance());
+    if (it == instances.end()) {
+      LOG_INFO << "Should not print this ";
```

Comment:
This should be an error stating we could not find the instance in question.

Reply:
This should never be printed. It is a failsafe check to leave some clue for debugging, if the user is to report some strange failure in the future. On the surface, the trigger for the print is "we could not find the instance in question", but in reality the instance should always be found, which "Should not print this" is a stronger indication on something is deeply wrong rather than something seems acceptable.